### DISCUSSION TITLE
Is there a way to ignore a specific line?

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#14916] [Request Help]
I have a list of dependencies for the same artifact, but with different versions. Renovate updates all of them:
https://github.com/TWiStErRob/net.twisterrob.gradle/pull/227/commits/575550ed788f8e901790017cc675a19e4d1404a0

I'm looking to say "ignore major updates for "com.android.tools.build:gradle 3.x, 4.x in this file only".

**Comment by viceice** [comment]
https://docs.renovatebot.com/configuration-options/#matchcurrentversion

**Comment by Fryie** [comment]
I'd really like this to be officially supported.
The context is that gradle allows to set dependency constraints, which are useful among other things to fix security vulnerabilities in transitive dependencies: https://docs.gradle.org/current/userguide/dependency_constraints.html

These are *lower* bounds on dependencies and it makes little sense for renovate to update them - worse, updating them without reason may break the parent dependency that is relying on a specific version of the transitive dependency.

Excluding that particular package globally is also not a very good option, because a particular dependency (say: `jackson-core`) can be used as a direct dependency for the project itself, but as a transitive dependency for the build script classpath.

There really should be an option to tell renovate "please ignore this version and don't try to update it". Unfortunately, I have no idea what kind of magic renovate uses to figure out where the version strings are that it needs to update. Even wrapping them in a map doesn't seem to help (this was working e.g. for dependabot, which isn't as "smart", apparently).

**Reply by RahulGautamSingh** [reply]
Use `packageRules` and `matchCurrentVersion`. Are you having problems with this setup?

**Reply by Fryie** [reply]
Matching only against a particular *dependency* is too limited because in a complex project, the same dependency can exist multiple times in different contexts. This is particularly common in the JVM landscape, especially with Gradle.

I have a gradle project at work that declares `jackson-core` as a *direct* compile-time dependency of the program. I want renovate to update this dependency automatically.

*At the same time*, the *build script* is using a gradle plugin that itself has `jackson-core` as a transitive dependency. Due to a recent CVE, I want to fix a minimum version for `jackson-core` using gradle's dependency constraints, but I do *not* want renovate to update this because it could cause the plugin to not work correctly anymore.

These two usages of `jackson-core` are completely distinct, they don't interact in any way and they can be and frequently are using different versions. This is something that renovate doesn't seem to support right now (unless I missed it).

In general, it doesn't make sense for renovate to want to update dependency constraints (as opposed to direct dependencies), and I should at least have the option to opt out of it in general. The only workaround that I've found so far is awkwardly splitting up constraints into separate files and excluding those, and this doesn't work very well for build script dependencies, due to the way gradle works.

**Reply by RahulGautamSingh** [reply]
How about capturing the dep at particular location with regex manager and then disabling it?

**Reply by Fryie** [reply]
can you explain how this would work exactly?

**Reply by RahulGautamSingh** [reply]
```
customManagers : [{
...// config to capture dep, 
// then give the dip a unique name using depNameTemplate
}],

packageRules: [{
..// capture unique depName using matchDepNames
enabled: false
}]
```
